compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs

```diff
@@ -626,30 +626,35 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
         let remove_entire_generics = num_redundant_args >= self.gen_args.args.len();
 
         let remove_lifetime_args = |err: &mut DiagnosticBuilder<'_>| {
-            let idx_first_redundant_lt_args = self.num_expected_lifetime_args();
-            let span_lo_redundant_lt_args =
-                self.gen_args.args[idx_first_redundant_lt_args].span().shrink_to_lo();
-            let span_hi_redundant_lt_args = self.gen_args.args
-                [idx_first_redundant_lt_args + num_redundant_lt_args - 1]
-                .span()
-                .shrink_to_hi();
-            let eat_comma =
-                idx_first_redundant_lt_args + num_redundant_lt_args - 1 != self.gen_args.args.len();
-
-            let span_redundant_lt_args = if eat_comma {
-                let span_hi = self.gen_args.args
-                    [idx_first_redundant_lt_args + num_redundant_lt_args - 1]
-                    .span()
-                    .shrink_to_hi();
-                span_lo_redundant_lt_args.to(span_hi)
-            } else {
-                span_lo_redundant_lt_args.to(span_hi_redundant_lt_args)
-            };
+            let mut lt_arg_spans = Vec::new();
+            let mut found_redundant = false;
+            for (i, arg) in self.gen_args.args.iter().enumerate() {
+                if let hir::GenericArg::Lifetime(_) = arg {
+                    if i > self.num_expected_lifetime_args() - 1 {
```

Comment:
```suggestion
                    if i >= self.num_expected_lifetime_args() {
```
Looks the case of 0 expected lifetime arguments is possible, the the current logic won't behave correctly.
(It would be good to add a test case for this.)

Reply:
Thank you for checking. And I also found that `lt_arg_spans.len()` instead of `i` should be used to get the number of lifetime arguments.